lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -108,6 +112,20 @@ public void OnNext(IActiveContext value)
             public void OnNext(IFailedEvaluator value)
             {
                 Logger.Log(Level.Error, FailedEvaluatorMessage);
+                if (value.FailedTask.Value == null)
+                {
+                    // TODO[JIRA REEF-1343]: fail the test if there's no failed task
+                    Logger.Log(Level.Error, "No failed task associated with failed evaluator");
+                }
+                else
+                {
+                    Logger.Log(Level.Error, "Failed task id '" + value.FailedTask.Value.Id + "'");
+                }
+            }
+            public void OnNext(ICompletedTask value)
+            {
+                // TODO[JIRA REEF-1343]: fail the test if receive ICompletedTask after failed evaluator
+                Logger.Log(Level.Info, "ICompletedTask");
```

Comment:
Looks like both FailedEvaluator and CompletedTask events are received , right? Would Adding ICompletedTask handler change the test result?